concurrency/runner.go

```diff
@@ -113,3 +117,33 @@ func ForEachJob(ctx context.Context, jobs int, concurrency int, jobFunc func(ctx
 	// Wait until done (or context has canceled).
 	return g.Wait()
 }
+
+// ForEachJobMergeResults is like ForEachJob but expects jobFunc to return a slice of results which are then
+// merged with results from all jobs. This function returns no results if an error occurred running any jobFunc.
+func ForEachJobMergeResults[T any, R any](ctx context.Context, jobs []T, concurrency int, jobFunc func(ctx context.Context, job T) ([]R, error)) ([]R, error) {
```

Comment:
nit: shall we rename `T` to `J` (as in Job)?

Reply:
Ah yes.

Reply:
Done in https://github.com/grafana/dskit/pull/486/commits/a880baf577deb957457f7711b720a53db60c2090.